plugin/src/main/groovy/com/kezong/fataar/ResourceModifier.groovy

```diff
@@ -116,98 +191,86 @@ class ResourceModifier {
                     if (!result.isValid) {
                         throw new RuntimeException(result.error)
                     }
-                } else if (attr.elements().size() > 0) {
-                    attrDef.enumDefinition = attr
+                } else {
+                    processAttrDefinition(attr, attrDef)
                 }
+                
+                attrDef.occurrences++
+                attrDef.declareStyleableElements.add(attr)
             }
         }
 
-        // Move only duplicate attrs to root level with merged formats
+        // Move duplicate attrs to root level if they're all in declare-styleable
         attrDefinitions.each { attrName, attrDef ->
-            if (attrDef.occurrences > 1) {
-                // Create or update root attr
-                Element rootAttr
-                List<Element> existingAttrs = root.elements("attr").findAll { it.attributeValue("name") == attrName }
-                if (existingAttrs.size() > 0) {
-                    rootAttr = existingAttrs.first()
-                    // Merge formats with existing
-                    String existingFormat = rootAttr.attributeValue("format")
-                    if (existingFormat) {
-                        if (attrDef.enumDefinition) {
-                            throw new RuntimeException(
-                                "Attribute '${attrName}' cannot mix enum/flag with other formats:\n" +
-                                "Definition with enum/flag: ${attrDef.enumDefinition.asXML()}\n" +
-                                "Attempted to merge with format: ${existingFormat}")
-                        }
-                        existingFormat.split("\\|").each { f -> attrDef.formats.add(f.trim()) }
-                    }
-                } else {
-                    rootAttr = attrDef.enumDefinition ? attrDef.enumDefinition.createCopy() : root.addElement("attr")
-                    rootAttr.addAttribute("name", attrName)
-                }
+            // Skip if not a duplicate
+            if (attrDef.occurrences <= 1) {
+                return
+            }
 
-                // Set merged format if any
-                if (!attrDef.formats.isEmpty()) {
-                    if (rootAttr.elements("enum").size() > 0 || rootAttr.elements("flag").size() > 0) {
-                        throw new RuntimeException(
-                            "Attribute '${attrName}' cannot mix enum/flag with other formats:\n" +
-                            "Definition with enum/flag: ${rootAttr.asXML()}\n" +
-                            "Attempted to merge with formats: ${attrDef.formats.join('|')}")
-                    }
-                    rootAttr.addAttribute("format", attrDef.formats.join("|"))
+            // Get first definition
+            Element firstDef = attrDef.getDefinition()
+            if (firstDef == null) {
+                return
+            }
+
+            // Check if any definition is in root
```

Comment:
if there is a definition in root, it should be the first one.
You can only check if the first element' parent is <declare-styleable>.